src/openrct2/ride/shops/Shop.cpp

```diff
@@ -33,7 +33,7 @@ static void PaintShop(
         return;
 
     CoordsXYZ offset(0, 0, height);
-    BoundBoxXYZ bb = { { 2, 2, height }, { 28, 28, 45 } };
+    BoundBoxXYZ bb = { { 2, 2, height }, { 28, 28, trackElement.GetClearanceZ() - trackElement.GetBaseZ() - 3 } };
```

Comment:
I'm talking about this, you are mixing visual bounds and clearance height, they are two separate things. This is something the object has to specify as bounds, clearance height is a separate thing.